### PR TITLE
Bugfix Guided Tour: Resize errors and positioning dots

### PR DESCRIPTION
**Pull request by lovaiible**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I added client tests (Jest) related to the features
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Description
This PR should fix several issues with the guided tour including https://github.com/ls1intum/Artemis/issues/943 and https://github.com/ls1intum/Artemis/issues/946 

- [x] The SourceTree setup tutorial video should be cropped and the noise in the audio should be removed
- [x] The tour should not be triggered on resize
- [x] The tour step should style should be responsive when resizing the browser window
- [x] The already executed step should hint should have a checkmark
- [x] The navigation dots on the tour step should be placed in the center.
- [x] The position of the dots should be calculated correctly when continuing the tour, see issue in screenshot:
image
- [x] The embedded video should not display related videos, see issue in screenshot: 
image

### Steps for Testing
1. Log in to Artemis
2. Navigate to Course Overview Page
3. Tour should be displayed automatically when it has not been visited before
4. Close tour and resize browser window
5. Tour should not be started automatically
6. When navigating through the tour, the dots should be displayed correctly

### Screenshots
Tour with browser resizing
1-min
Tour does not trigger on resize

Retested success tour

Retested fail tour

No suggested videos when pausing or finishing the video
image

### CODE REVIEW

File: src/main/webapp/app/guided-tour/guided-tour.service.ts [hunk 1]
```diff
@@ -616,11 +616,12 @@ export class GuidedTourService {
          * to prevent ExpressionChangedAfterItHasBeenCheckedError
          */
         setTimeout(() => {
-            this.currentTour = cloneDeep(guidedTour);
```
Comment (lovaiible) [inline src/main/webapp/app/guided-tour/guided-tour.service.ts]: `this.currentTour` is checked when resizing the browser. The bug here was the it was set even before we know if the tour should be started automatically or not.

File: src/main/webapp/app/guided-tour/guided-tour.component.html [hunk 2]
```diff
@@ -37,6 +37,14 @@ <h5 jhiTranslate="{{ currentTourStep.subHeadlineTranslateKey }}" class="sub-head
                         <div [innerHTML]="currentTourStep.hintTranslateKey | translate | htmlForGuidedTourMarkdown"></div>
                     </div>
                 </div>
+                <div *ngIf="currentTourStep.alreadyExecutedTranslateKey" class="step-hint interaction alert alert-success">
```
Comment (lovaiible) [inline src/main/webapp/app/guided-tour/guided-tour.component.html]: I would refactor all the hints into a new `guided-tour-hints` component in a future PR

File: src/main/webapp/i18n/de/guidedTourCourseExerciseOverview.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
                 "sourceTreeSetup": {
                     "headline": "Konfiguriere SourceTree",
                     "content": "Schau dir bitte das Video an und folge den Schritten, um SourceTree richtig aufzusetzen. Mit diesem Schritt haben wir das Setup abgeschlossen und können uns die Kursseite und dessen Elemente genauer unter die Lupe nehmen.",
-                    "videoUrl": "https://www.youtube.com/embed/zVw-mMgQjPU",
+                    "videoUrl": "https://www.youtube.com/embed/zVw-mMgQjPU?rel=0",
```
Comment (lovaiible) [inline src/main/webapp/i18n/de/guidedTourCourseExerciseOverview.json]: The `?rel=0` param is used so that the related videos are not displayed

**Comment by lovaiible** [comment]
https://bamboobruegge.in.tum.de/browse/ARTEMIS-WEBAPP738-14
should be used for testing on the test server

File: src/main/webapp/app/guided-tour/guided-tour.service.ts [hunk 4]
```diff
@@ -691,11 +700,12 @@ export class GuidedTourService {
         if (currentIndex < nextIndex) {
             // Moves the n-small and p-small class one dot further
             if (nextDot && nextDot.classList.contains('n-small') && lastDot && !lastDot.classList.contains('n-small')) {
-                this.transformCount += transformXIntervalNext;
+                this.transformCount += this.transformXIntervalNext;
                 nextDot.classList.remove('n-small');
                 nextPlusOneDot.classList.add('n-small');
                 dotList.style.transform = 'translateX(' + this.transformCount + 'px)';
                 dotList.querySelectorAll('li').forEach((node, index) => {
+                    console.log('nextIndex - 8: ', nextIndex - 8);
```
Comment (krusche) [inline src/main/webapp/app/guided-tour/guided-tour.service.ts]:
```suggestion
```

Comment (krusche) [reply src/main/webapp/app/guided-tour/guided-tour.service.ts]: Remove console log

**Pull request merged**